pts-core/objects/phodevi/components/phodevi_system.php

```diff
@@ -1018,7 +1018,28 @@ public static function sw_kernel_date()
 	}
 	public static function sw_kernel()
 	{
-		return php_uname('r');
+		if(phodevi::is_windows())
+		{
+			// CurrentBuild and CurrentVersion are available since at least NT 4.0
+			// CurrentVersion is frozen at 6.3 (same as Windows 8.1) in Windows 10 & 11
+			$current_build = trim(shell_exec('powershell "If (Get-ItemProperty -ErrorAction SilentlyContinue -Path \'Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\' CurrentBuild) { (Get-ItemProperty -Path \'Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\' CurrentBuild).CurrentBuild } Else { $null }"'));
+			$current_version = trim(shell_exec('powershell "If (Get-ItemProperty -ErrorAction SilentlyContinue -Path \'Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\' CurrentVersion) { (Get-ItemProperty -Path \'Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\' CurrentVersion).CurrentVersion } Else { $null }"'));
```

Comment:
As powershell calls can be expensive due to often measurable startup times, probably wise moving the $current_version line inside the else where it's only used to avoid otherwise making the call if not being used.

Reply:
Shuffled things around to avoid more calls than is necessary.